ocaml/xapi/xapi_vm.ml

```diff
@@ -249,21 +249,30 @@ let start ~__context ~vm ~start_paused ~force =
   let default_value = match Xapi_xenops.firmware_of_vm vmr with
     | Bios -> Vm_platform.fallback_device_model_default_value
     | Uefi _ -> Vm_platform.fallback_device_model_default_value_uefi in
-  let platform = vmr.API.vM_platform
+  let platform = ref( vmr.API.vM_platform
                  |> Xapi_vm_helpers.ensure_device_model_profile_present
                    ~__context ~domain_type:vmr.API.vM_domain_type
-                   ~default_value ~is_a_template:false
+                   ~default_value ~is_a_template:false)
   in
   let vmr =
-    if platform <> vmr.API.vM_platform then begin
-      Db.VM.set_platform ~__context ~self:vm ~value:platform;
-      { vmr with API.vM_platform = platform }
+    if !platform <> vmr.API.vM_platform then begin
+      platform := !platform;
+      { vmr with API.vM_platform = !platform }
     end else vmr
   in
 
+  (* Check to see if secure boot should be parsed *)
+  let pool = Helpers.get_pool ~__context in
+  if List.assoc "secureboot" !platform = "auto" then
+    platform := (("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self:pool <> "")) :: List.remove_assoc "secureboot" !platform);
+  Db.VM.set_platform ~__context ~self:vm ~value:!platform;
 
-  (* Check to see if we're using correct device-model when vm has VUSBs*)
+  if not (Array.exists (fun e -> e = "PK.auth") (Sys.readdir "/usr/share/varstored")) then
+    let open_file = open_out "/usr/share/varstored/xapi_uefi_certificates.tar" in
+    finally (fun () -> fprintf open_file "%s" (Xapi_stdext_base64.Base64.decode (Db.Pool.get_uefi_certificates ~__context ~self:pool)))
```

Comment:
How long is the string? Why not write the string to `open_file` without using `fprintf` (which adds nothing)? if writint fails, should it be logged?